src/epub.rs

```diff
@@ -27,12 +27,52 @@ pub enum EpubVersion {
     V30,
 }
 
+/// The page-progression-direction attribute of spine is a global attribute and
+/// therefore defines the pagination flow of the book as a whole.
+#[derive(Debug, Copy, Clone, Default)]
+pub enum PageDirection {
+    /// Left to right
+    #[default]
+    Ltr,
+    /// Right to left
+    Rtl,
+}
+
+impl ToString for PageDirection {
+    fn to_string(&self) -> String {
+        match &self {
+            PageDirection::Rtl => "rtl".into(),
+            PageDirection::Ltr => "ltr".into(),
+        }
+    }
+}
+
+impl From<String> for PageDirection {
+    fn from(value: String) -> Self {
+        value.parse().unwrap()
```

Comment:
It is still not a good design to have partial functions like this. See also the [`From` docs](https://doc.rust-lang.org/std/convert/trait.From.html):

> **Note: This trait must not fail.** The From trait is intended for perfect conversions. If the conversion can fail or is not perfect, use [TryFrom](https://doc.rust-lang.org/std/convert/trait.TryFrom.html).

Just use `from_str` in `metadata` function and propagate the error when it fails.